client/my-sites/plan-features-2023-grid/components/billing-timeframe.tsx

```diff
@@ -88,12 +97,83 @@ function usePerMonthDescription( {
 	return null;
 }
 
+interface WooExpressMonthlyPromotionProps {
+	billingTimeframe: TranslateResult;
+	monthlyPlanSlug: string;
+	yearlyPlanSlug: string;
+}
+
+const WooExpressMonthlyPromotion: FunctionComponent< WooExpressMonthlyPromotionProps > = (
+	props
+) => {
+	const translate = useTranslate();
+	const { billingTimeframe, yearlyPlanSlug, monthlyPlanSlug } = props;
+
+	const annualPlan = getPlans()[ yearlyPlanSlug ];
+	const monthlyPlan = getPlans()[ monthlyPlanSlug ];
+	const planPrices = useSelector( ( state ) => ( {
+		annualPlanMonthlyPrice: getPlanRawPrice( state, annualPlan.getProductId(), true ) || 0,
+		monthlyPlanPrice: getPlanRawPrice( state, monthlyPlan.getProductId() ) || 0,
+	} ) );
+
+	const percentageSavings = Math.round(
+		( 1 - planPrices.annualPlanMonthlyPrice / planPrices.monthlyPlanPrice ) * 100
+	);
+	const discountDescription = percentageSavings
+		? translate( `Save %(percentageSavings)s%% by paying annually`, {
+				args: {
+					percentageSavings,
+				},
+				comment: 'Translators: percentageSavings is the percentage saved by paying annually',
+		  } )
+		: null;
+	return (
+		<div>
+			<div>{ billingTimeframe }</div>
+			<div className="plan-features-2023-grid__discount-promotion">{ discountDescription }</div>
+		</div>
+	);
+};
+
+/**
+ * Retrieves the monthly and yearly plan slugs for a given WooExpress plan name.
+ *
+ * @param planName - The name of the WooExpress plan to retrieve slugs for.
+ * @returns An object containing the monthly and yearly plan slugs, or an empty object if the plan name is invalid.
+ */
+function getWooExpressPlanSlugs( planName: string ) {
```

Comment:
Added a note about this above.

Reply:
Do you think we could tackle that in a follow-up PR/issue? I'm not sure yet how adding those types can impact the Woo Express flows.

Reply:
I started working on a draft PR here: https://github.com/Automattic/wp-calypso/pull/76535